src/approveAndCall.ts

```diff
@@ -0,0 +1,16 @@
+import { Interface } from '@ethersproject/abi'
+import { abi } from '@uniswap/swap-router-contracts/artifacts/contracts/interfaces/IApproveAndCall.sol/IApproveAndCall.json'
+import { Token } from '@uniswap/sdk-core'
+
+export abstract class ApproveAndCall {
```

Comment:
probably should add `callPositionManager` into this interface

i wonder whether it should accept an array of calls, which would be passed on directly if length == 1, and otherwise formatted against the nft `multicall`? i kinda think latter?